gdb-7.8/gdb/breakpoint.c

```diff
@@ -13614,6 +13617,40 @@ bkpt_resources_needed (const struct bp_location *bl)
 }
 
 static enum print_stop_action
+gpu_bkpt_trigger_print_it (bpstat bs)
+{
+  struct breakpoint *b;
+  const struct bp_location *bl;
+  int bp_temp;
+  struct ui_out *uiout = current_uiout;
+
+  gdb_assert (bs->bp_location_at != NULL);
+
+  bl = bs->bp_location_at;
+  b = bs->breakpoint_at;
+
+  bp_temp = b->disposition == disp_del;
+  if (bl->address != bl->requested_address)
+    breakpoint_adjustment_warning (bl->requested_address,
+				   bl->address,
+				   b->number, 1);
+  annotate_breakpoint (b->number);
+  if (bp_temp)
+    ui_out_text (uiout, "\nTemporary breakpoint triggered by GPU Kernel Breakpoint");
+  else
+    ui_out_text (uiout, "\nStopped on GPU Breakpoint");
```

Comment:
If we add PRINT_NOTHING below, then a new line is needed in 
ui_out_text (uiout, "\nStopped on GPU Breakpoint\n");
so that the (rocm-gdb) prompt comes on a new line.

also," ui_out_text (uiout, ", ");" should be removed.